src/app/modules/shopping-list/pages/add-products-shopping-list/add-products-shopping-list.component.ts

```diff
@@ -105,6 +105,10 @@ export class AddProductsShoppingListComponent implements OnInit, AfterContentIni
     disabled: signal(false),
     render: signal(false),
     click: () => {
+      if (!this.showMoreButton.disabled()) {
+        // Call the showMore method to load additional products
+        this.showMoreButton.click();
+      }
```

Comment:
I don't understand this change, why do you call the `click` method again?, if not because I replace this function below, this change causes an infinite loop.

Reply:
Replaced the recursive call with a direct reference to the method this.algoliaService.showMoreProducts()

Reply:
The change causes the application not to start.

Reply:
but i just reverse everything as it was before and click function is empty

Reply:
You have set the “click” property to null

![image](https://github.com/user-attachments/assets/b2cacae3-2a8a-4ede-930d-c3856b1dd922)